arduino/waveform/Waveform.cpp

```diff
@@ -0,0 +1,41 @@
+#include "Waveform.h"
+
+Waveform::Waveform(Cyclops *_cyclops, Source *_source) : 
+	status(INIT)
```

Comment:
I would just define status in the header instead of using using the initialization list since its value is not dependent on the constructor args.